### PR TITLE
Add lightfm and implicit training examples

### LINKED ISSUE #561: [Task] Example of training Implicit in Merlin Models
### Description
Add an example to train a model with implicit library with Merlin Models:
https://github.com/NVIDIA-Merlin/Merlin/issues/104

### LINKED ISSUE #562: [Task] Example of training LightFM in Merlin Models
### Description
Add an example to train LightFM with Merlin Models

### PR DESCRIPTION
**Pull request by radekosmulski**
resolves #562, resolves #561

### CODE REVIEW

**Comment by radekosmulski** [comment]
rerun tests

File: tests/integration/tf/test_ci_07_third_party_integration.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    REPO_ROOT / "examples/07-Train-a-third-party-model-using-the-Merlin-Models-API.ipynb",
+    execute=False,
+)
+def test_func(tb):
+    tb.inject(
+        """
+        import os
+        os.environ["INPUT_DATA_DIR"] = "/raid/data/movielens"
+        """
+    )
```
Comment (bschifferer) [inline tests/integration/tf/test_ci_07_third_party_integration.py:16]: I dont think this works in the same way than the movielens notebooks.

In https://github.com/NVIDIA-Merlin/models/blob/main/examples/01-Getting-started.ipynb we have a variable
```
input_path = os.environ.get("INPUT_DATA_DIR", os.path.expanduser("~/merlin-models-data/movielens/"))
train, valid = get_movielens(variant="ml-1m", path=input_path)
```

The CI machine has following data available
```
    |-- movielens-1m
    |   |-- README
    |   |-- movies.dat
    |   |-- ratings.dat
    |   `-- users.dat
    |-- movielens-25m
    |   |-- README.txt
    |   |-- genome-scores.csv
    |   |-- genome-tags.csv
    |   |-- links.csv
    |   |-- movies.csv
    |   |-- ratings.csv
    |   `-- tags.csv
```

This notebooks doesnt use INPUT_DATA_DIR
```
train, valid = get_movielens(variant='ml-100k')
```

My suggestions is:
- let's remove the integration tests for now, that we can merge the example. I think it is fine, if the unittest covers it.
- Create a ticket to add integration tests, that we do not forget it
- Talk to @jperez999 to add movielens 100k to the integration test machine.

The goal of integration tests are that it doesnt require to download the dataset everytime and runs the CI with realdata. So we should uplod the data into the correct directory and update the example with correct paths

**Review by bschifferer — CHANGES_REQUESTED** [review]

File: examples/07-Train-a-third-party-model-using-the-Merlin-Models-API.ipynb [hunk 2]
```diff
@@ -2,7 +2,7 @@
  "cells": [
```
Comment (bschifferer) [inline examples/07-Train-a-third-party-model-using-the-Merlin-Models-API.ipynb:2]: <p>Can we rename the notebooks and all reference to <code> traditional ml models</code> . We are leveraging 3rd party libraries, it sounds not well. TensorFlow is a 3rd party library as well.</p><p> </p><p>Can you have the same order as in the notebook:  Change <code>xgboost</code>, <code>implicit</code> and <code>lightFM</code> to XGBoost, lightFM, implicit</p><p> </p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/NVIDIA-Merlin/models/pull/629/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (bschifferer) [inline examples/07-Train-a-third-party-model-using-the-Merlin-Models-API.ipynb:2]: <p>Can we rename it to Training a XGBoost model - similar to the other headings</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/NVIDIA-Merlin/models/pull/629/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

File: tests/integration/tf/test_ci_07_third_party_integration.py (hunk 1 above)
Comment (radekosmulski) [reply tests/integration/tf/test_ci_07_third_party_integration.py:16]: Makes a lot of sense! 🙂 Will follow up with @jperez999, he is on holiday atm so asked on the ci channel, maybe someone will be able to help.

Removed the test for the time being and created an issue here

Task created #639

File: examples/07-Train-a-third-party-model-using-the-Merlin-Models-API.ipynb (hunk 2 above)
Comment (radekosmulski) [reply examples/07-Train-a-third-party-model-using-the-Merlin-Models-API.ipynb:2]: done

Comment (radekosmulski) [reply examples/07-Train-a-third-party-model-using-the-Merlin-Models-API.ipynb:2]: done

**Review by bschifferer — APPROVED** [review]

**Comment by rnyak** [comment]
do we want to have GPU tests for implicit and LightFM models?

**Comment by rnyak** [comment]
do we want to add GPU testing of `implicit` and `lightfm`? shall we addd the requirements to CI image for that? Or shall we go with CPU testing for now and rely on the GH actions?

**Comment by benfred** [comment]
rerun tests

**Comment by benfred** [comment]
@bschifferer I see that the implicit/lightfm tests are no longer skipped in the jenkins runs (though I am unsure why they were skipped in the previous run 5 days ago)

**Comment by karlhigley** [comment]
The CI runner image didn't get successfully built until Friday last week, so I'm guessing the new packages hadn't been part of a successful CI runner build yet

**Comment by radekosmulski** [comment]
rerun tests

**Pull request merged**